module/synchronization/core.go

```diff
@@ -356,7 +382,8 @@ func (c *Core) RangeRequested(ran flow.Range) {
 
 // BatchRequested updates status state for a batch of block IDs that has been
 // successfully requested. Must be called when a batch request is submitted.
-func (c *Core) BatchRequested(batch flow.Batch) {
+func (c *Core) BatchRequested(batch chainsync.Batch) {
+	c.metrics.BatchRequested(batch)
 	c.mu.Lock()
 	defer c.mu.Unlock()
```

Comment:
better to move after the lock
```suggestion
	c.mu.Lock()
	defer c.mu.Unlock()
	c.metrics.BatchRequested(batch)
```